gossipsub.go

```diff
@@ -434,12 +619,20 @@ func (gs *GossipSubRouter) Join(topic string) {
 
 	gmap, ok = gs.fanout[topic]
 	if ok {
+		// these peers have a score above the publish threshold, which may be negative
+		// so drop the ones with a negative score
+		for p := range gmap {
+			if gs.score.Score(p) < 0 {
+				delete(gmap, p)
+			}
+		}
```

Comment:
Why do this _here_? To avoid having to eagerly maintain `gs.fanout`?

Reply:
yes indeed.